libs/ui/src/ui_strings/audio_context.tsx

```diff
@@ -3,28 +3,136 @@ import React from 'react';
 import { Optional } from '@votingworks/basics';
 
 import { UiStringsReactQueryApi } from '../hooks/ui_strings_api';
+import {
+  DEFAULT_PLAYBACK_RATE,
+  MAX_PLAYBACK_RATE,
+  MIN_PLAYBACK_RATE,
+  PLAYBACK_RATE_INCREMENT_AMOUNT,
+} from './audio_playback_rate';
+import {
+  GAIN_INCREMENT_AMOUNT_DB,
+  DEFAULT_GAIN_DB,
+  MAX_GAIN_DB,
+  MIN_GAIN_DB,
+} from './audio_volume';
 
-export interface AudioContextInterface {
-  // TODO(kofi): Flesh out.
+export interface UiStringsAudioContextInterface {
+  api: UiStringsReactQueryApi;
+  decreasePlaybackRate: () => void;
+  decreaseVolume: () => void;
+  gainDb: number;
+  increasePlaybackRate: () => void;
+  increaseVolume: () => void;
+  isEnabled: boolean;
+  playbackRate: number;
+  reset: () => void;
+  setIsEnabled: (enabled: boolean) => void;
+  togglePause: () => void;
+  webAudioContext?: AudioContext;
 }
 
-const AudioContext =
-  React.createContext<Optional<AudioContextInterface>>(undefined);
+export const UiStringsAudioContext =
+  React.createContext<Optional<UiStringsAudioContextInterface>>(undefined);
 
-export function useAudioContext(): Optional<AudioContextInterface> {
-  return React.useContext(AudioContext);
+export function useAudioContext(): Optional<UiStringsAudioContextInterface> {
+  return React.useContext(UiStringsAudioContext);
 }
 
-export interface AudioContextProviderProps {
-  // eslint-disable-next-line react/no-unused-prop-types
+export interface UiStringsAudioContextProviderProps {
   api: UiStringsReactQueryApi;
   children: React.ReactNode;
 }
 
-export function AudioContextProvider(
-  props: AudioContextProviderProps
+export function UiStringsAudioContextProvider(
+  props: UiStringsAudioContextProviderProps
 ): JSX.Element {
-  const { children } = props;
+  const { api, children } = props;
+  const [isEnabled, setIsEnabled] = React.useState(false);
+  const [isPaused, setIsPaused] = React.useState(true);
+  const [playbackRate, setPlaybackRate] = React.useState<number>(
+    DEFAULT_PLAYBACK_RATE
+  );
+  const [gainDb, setGainDb] = React.useState<number>(DEFAULT_GAIN_DB);
+
+  const webAudioContextRef = React.useRef(
+    window.AudioContext ? new AudioContext() : undefined
```

Comment:
I'm guessing this isn't an expensive operation, but I thought I'd ask just in case: is creating a new `AudioContext` expensive? Looking at the [Chromium source code for `AudioContext`](https://chromium.googlesource.com/chromium/blink/+/refs/heads/main/Source/modules/webaudio/AudioContext.cpp#28) it appears that it might spawn a new thread at the very least. Can we use a `useMemo` for this maybe?

Reply:
Yeah, while I was prototyping, it registered as a separate JS VM instance using up somewhere between 1-1.5mb (whenever there's a fully decoded clip playing), so not too bad.

The intention's to only ever have one instance per app load, though. Previously had this as a lazy-instantiated single global instance and moved it to a `ref` to make testing easier, but your comment makes me question my understanding of how often `ref`s are re-instantiated. Will check and maybe move back to the singleton

Reply:
Agreed, that's not terrible. And it looks like the source hardcodes a max of 6 and throws an exception if it goes over, so we'd become aware of it if we were creating too many that weren't being cleaned up.

As for `ref`, that doesn't really have anything to do with it afaict. Whenever the component is rendered it'll create a new `AudioContext` which is passed to `useRef` and promptly dropped. So we'd pay the startup+teardown costs with every render. Normally I'd say it's not worth thinking about e.g. creating some JS data structure repeatedly only to have it be gc'd, but this might do significantly more so I'm glad you opened #4478 to fix this 🙏